app/src/main/java/org/piwigo/ui/settings/SettingsViewModel.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Piwigo for Android
+ * Copyright (C) 2019-2019 Piwigo Team http://piwigo.org
+ * Copyright (C) 2019-2019 Radko Varchola
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.piwigo.ui.settings;
+
+import androidx.lifecycle.ViewModel;
+
+public class SettingsViewModel extends ViewModel {
```

Comment:
you closed this conversation without answering the question? What is the use of this empty ViewModel? Is a ViewModel generally used for PreferenceActivities or what is recommended here?

Reply:
@ramack  Sorry i closed it without answer. I am little lost between this comments. I will remove it and try if everything works correctly without it. I hope DI will be working correctly. Because we injecting PreferenceRepository in SettingsActivity.